tests/unit/igraph_is_forest.c

```diff
@@ -0,0 +1,95 @@
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include <igraph.h>
+
+#include "test_utilities.inc"
+
+void check_output(const igraph_t *graph,igraph_bool_t *res,igraph_neimode_t mode){
+    igraph_bool_t result;
+    igraph_vector_t roots;
+    igraph_vector_init(&roots, 0);
+    IGRAPH_FINALLY(igraph_vector_destroy, &roots);
+    igraph_is_forest(graph,&result,&roots, mode);
+    IGRAPH_ASSERT(*res==result);
+    if(result)
+    igraph_vector_print(&roots);
+    igraph_vector_destroy(&roots);
+}
+
+int main() {
+    igraph_t graph;
+    igraph_bool_t res;
+    igraph_neimode_t mode;
+
+    /*Empty Graph*/
```

Comment:
If you print something before each test you can see which output belongs to which test. Else the test results are hard to check.